src/main/java/org/gridsuite/shortcircuit/server/ShortCircuitController.java

```diff
@@ -44,31 +44,21 @@ public ShortCircuitController(ShortCircuitService shortCircuitService) {
-    @ApiResponses(value = {@ApiResponse(responseCode = "200",
-                                        description = "The short circuit analysis has been performed",
-                                        content = {@Content(mediaType = APPLICATION_JSON_VALUE,
-                                                            schema = @Schema(implementation = ShortCircuitParameters.class))})})
+    @ApiResponse(responseCode = "200", description = "The short circuit analysis has been performed")
     public ResponseEntity<UUID> runAndSave(@Parameter(description = "Network UUID") @PathVariable("networkUuid") UUID networkUuid,
                                            @Parameter(description = "Variant Id") @RequestParam(name = "variantId", required = false) String variantId,
                                            @Parameter(description = "Result receiver") @RequestParam(name = "receiver", required = false) String receiver,
                                            @Parameter(description = "reportUuid") @RequestParam(name = "reportUuid", required = false) UUID reportUuid,
                                            @Parameter(description = "reporterId") @RequestParam(name = "reporterId", required = false) String reporterId,
                                            @Parameter(description = "The type name for the report") @RequestParam(name = "reportType", required = false) String reportType,
                                            @Parameter(description = "Bus Id - Used for analysis targeting one bus") @RequestParam(name = "busId", required = false) String busId,
-                                           @RequestBody(required = false) ShortCircuitParameters parameters,
+                                           @Parameter(description = "ID of parameters to use") @RequestParam(name = "parametersUuid", required = false) Optional<UUID> parametersUuid,
+                                           @Parameter(description = "Parameters (full or partial) to use on top of the ones designed by parametersUuid", schema = @Schema(implementation = ShortCircuitParameters.class))
+                                           @RequestBody(required = false) Optional<ShortCircuitParametersInfos> parameters,
                                            @RequestHeader(HEADER_USER_ID) String userId) {
-        ShortCircuitParameters nonNullParameters = getNonNullParameters(parameters);
-        ShortCircuitRunContext runContext = new ShortCircuitRunContext(networkUuid, variantId, receiver, nonNullParameters, reportUuid, reporterId, reportType, userId, null, busId);
-        UUID resultUuid = shortCircuitService.runAndSaveResult(runContext);
-        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(resultUuid);
+        return ResponseEntity.ok().contentType(TEXT_PLAIN).body(shortCircuitService.runAndSaveResult(networkUuid, variantId, receiver, reportUuid, reporterId, reportType, userId, busId, parametersUuid, parameters));
```

Comment:
Why do you change that ?

Reply:
I need to override the parameters, I'm doing it in the service (It's out of controller scope).